syntax = "proto3";
package finnews;

// v2 - pass source/url through ArticleEnriched

message ArticleRaw {
  string id = 1;           // stable hash of (source, url) or GUID
  string source = 2;        // e.g., "Reuters", "CompanyPR"
  string url = 3;
  string title = 4;
  string body = 5;          // optional at raw stage
  int64 published_ts = 6;   // epoch ms
  int64 ingested_ts = 7;
}

message ArticleClean {
  string id = 1;
  string source = 2;
  string url = 3;
  string title = 4;
  string body = 5;           // cleaned text, boilerplate removed
  int64 published_ts = 6;
  repeated string hints = 7; // cashtags, domains, path tokens
  string language = 8;
}

message ArticleEnriched {
  string id = 1;
  string title = 2;
  string body = 3;
  int64 published_ts = 4;
  repeated string tickers = 5;       // ["AAPL","MSFT"]
  repeated string entities = 6;      // org names
  repeated string topics = 7;        // preliminary rule-based tags
  string source = 8;                 // NEW: passthrough (from ArticleClean.source)
  string url = 9;                    // NEW: passthrough (from ArticleClean.url)
}

message ArticleScored {
  string id = 1;
  repeated string tickers = 2;
  float sentiment = 3;               // -1..1
  repeated string topics = 4;        // final topics from model (and/or merged)
  float confidence = 5;
  int64 scored_ts = 6;
}
